packages/credential-store/src/agent/CredentialStore.ts

```diff
@@ -0,0 +1,257 @@
+import { AbstractDigitalCredentialStore } from '@sphereon/ssi-sdk.data-store/dist/digitalCredential/AbstractDigitalCredentialStore'
+import { IVerifiableCredential } from '@sphereon/ssi-types'
+
+// Exposing the methods here for any REST implementation
+export const credentialStoreMethods: Array<string> = [
+  'crsAddCredential',
+  'crsUpdateCredentialState',
+  'crsGetCredential',
+  'crsGetCredentials',
+  'crsStoreCredential',
+  'crsDeleteCredential',
+  'crsDeleteCredentials',
+  'crsGetCredentialsByClaims',
+  'crsGetCredentialsByClaimsCount',
+]
+
+/**
+ * {@inheritDoc ICRManager}
+ */
+export class CredentialStore implements IAgentPlugin {
+  readonly schema = schema.ICredentialStore
+  readonly methods: ICredentialStore = {
+    crsAddCredential: this.crsAddCredential.bind(this),
+    crsUpdateCredentialState: this.crsUpdateCredentialState.bind(this),
+    crsGetCredential: this.crsGetCredential.bind(this),
+    crsGetCredentials: this.crsGetCredentials.bind(this),
+    crsGetUniqueCredentialByIdOrHash: this.crsGetUniqueCredentialByIdOrHash.bind(this),
+    crsGetUniqueCredentials: this.crsGetUniqueCredentials.bind(this),
+    crsDeleteCredential: this.crsDeleteCredential.bind(this),
+    crsDeleteCredentials: this.crsDeleteCredentials.bind(this),
+    crsGetCredentialsByClaims: this.crsGetCredentialsByClaims.bind(this),
+    crsGetCredentialsByClaimsCount: this.crsGetCredentialsByClaimsCount.bind(this),
+  }
+
+  private readonly store: AbstractDigitalCredentialStore
+
+  constructor(options: { store: AbstractDigitalCredentialStore }) {
+    this.store = options.store
+  }
+
+  /** {@inheritDoc ICRManager.crmAddCredential} */
+  private async crsAddCredential(args: AddCredentialArgs): Promise<DigitalCredential> {
+    return await this.store.addCredential(args.credential)
+  }
+
+  /** {@inheritDoc ICRManager.updateCredentialState} */
+  private async crsUpdateCredentialState(args: UpdateCredentialStateArgs): Promise<DigitalCredential> {
+    return await this.store.updateCredentialState(args)
+  }
+
+  /** {@inheritDoc ICRManager.crmGetCredential} */
+  private async crsGetCredential(args: GetCredentialArgs): Promise<DigitalCredential> {
+    const { id } = args
+    const credential = await this.store.getCredential({ id })
+    return credential
+  }
+
+  /** {@inheritDoc ICRManager.crmGetCredentials} */
+  private async crsGetCredentials(args: GetCredentialsArgs): Promise<Array<DigitalCredential>> {
+    const { filter } = args
+    const credentials = await this.store.getCredentials({ filter })
+    return credentials.data
+  }
+
+  /** {@inheritDoc ICRManager.crmGetUniqueCredentialByIdOrHash} */
+  private async crsGetUniqueCredentialByIdOrHash(args: GetCredentialsByIdOrHashArgs): Promise<OptionalUniqueDigitalCredential> {
+    const credentials = await this.crsGetCredentials({ filter: credentialIdOrHashFilter(args.credentialRole, args.idOrHash) })
+    if (credentials.length === 0) {
+      return undefined
+    }
```

Comment:
Probably wise to log in case length > 1.
I can imagine this would happen if the issuer, holder and rp are using the same store for the same credential, but each in their own context. Then we end up with 3 versions of the same credential in the store, but with different roles.

Reply:
Add logger.warning